src/Http/Http/test/ResponseCookiesTest.cs

```diff
@@ -54,6 +54,168 @@ public void AppendSameSiteNoneWithoutSecureLogsWarning()
+        ]);
+    }
+
+    [Fact]
+    public void AppendPartitionedLogsWarningsForEachCookie()
+    {
+        var headers = (IHeaderDictionary)new HeaderDictionary();
+        var features = MakeFeatures(headers);
+        var services = new ServiceCollection();
+
+        var sink = new TestSink(TestSink.EnableWithTypeName<ResponseCookies>);
+        var loggerFactory = new TestLoggerFactory(sink, enabled: true);
+        services.AddLogging();
+        services.AddSingleton<ILoggerFactory>(loggerFactory);
+
+        features.Set<IServiceProvidersFeature>(new ServiceProvidersFeature() { RequestServices = services.BuildServiceProvider() });
+
+        var cookies = new ResponseCookies(features);
+        var testCookie1 = "TestCookie1";
+        var testCookie2 = "TestCookie2";
+
+        var cookieDict = new[]
+        {
+            new KeyValuePair<string, string>(testCookie1, "value1"),
+            new KeyValuePair<string, string>(testCookie2, "value2"),
+        };
+
+        cookies.Append(cookieDict, new CookieOptions()
+        {
+            Partitioned = true,
+            // Missing SameSite = SameSiteMode.None,
+            // Missing Secure = true,
+        });
+
+        var cookieHeaderValues = headers.SetCookie;
+        Assert.All(headers.SetCookie, cookieHeaderValue =>
+        {
+            Assert.Contains("partitioned", cookieHeaderValue);
+            Assert.DoesNotContain("secure", cookieHeaderValue);
+            Assert.DoesNotContain("samesite", cookieHeaderValue);
+        });
+
+        Assert.Collection(sink.Writes,
+        [
+            entry => Assert.Equal($"The cookie '{testCookie1}' has set 'Partitioned' and must also set 'Secure'.", entry.Message),
+            entry => Assert.Equal($"The cookie '{testCookie1}' has set 'Partitioned' and should also set 'SameSite=None'.", entry.Message),
+            entry => Assert.Equal($"The cookie '{testCookie2}' has set 'Partitioned' and must also set 'Secure'.", entry.Message),
+            entry => Assert.Equal($"The cookie '{testCookie2}' has set 'Partitioned' and should also set 'SameSite=None'.", entry.Message),
+        ]);
+    }
+
+    [Fact]
+    public void AppendPartitionedCorrectlyDoesNotLog()
+    {
+        var headers = (IHeaderDictionary)new HeaderDictionary();
+        var features = MakeFeatures(headers);
+        var services = new ServiceCollection();
+
+        var sink = new TestSink(TestSink.EnableWithTypeName<ResponseCookies>);
+        var loggerFactory = new TestLoggerFactory(sink, enabled: true);
+        services.AddLogging();
+        services.AddSingleton<ILoggerFactory>(loggerFactory);
+
+        features.Set<IServiceProvidersFeature>(new ServiceProvidersFeature() { RequestServices = services.BuildServiceProvider() });
+
+        var cookies = new ResponseCookies(features);
+        var testCookie = "TestCookie";
+
+        cookies.Append(testCookie, "value", new CookieOptions()
+        {
+            Partitioned = true,
```

Comment:
Can we add a test that sets this via the `CookieBuilder`? [CookieTests.CookieOptionsAlterSetCookieHeader](https://github.com/dotnet/aspnetcore/blob/eb18cb23cf2be9db70ae799e9b23a74a26e78d0b/src/Security/Authentication/test/CookieTests.cs#L356) for the CookieAuthenticationHandler is probably a good place to add this since this already tests a bunch of options on `CookieAuthenticationOptions.Cookie` which is a `CookieBuilder`.

Reply:
Sure.

Reply:
@halter73 Do we want the [correlation cookie](https://github.com/dotnet/aspnetcore/blob/c157c5c1962a66c4fe0212ed767f167bb9cdfb79/src/Security/Authentication/Core/src/RemoteAuthenticationOptions.cs#L147) to be partitioned by default?